geocase-linkage/main.py

```diff
@@ -21,91 +18,76 @@ def start_kafka() -> None:
-            send_updated_opends(mas_job_record, producer)
+            mas_job_record = map_to_annotation_event(specimen_data, result,
+                                                     json_value['jobId'])
+            publish_annotation_event(mas_job_record, producer)
         except Exception as e:
             logging.exception(e)
 
 
-def map_to_mas_job_record(specimen_data: Dict, results: List[Dict[str, str]], job_id: str) -> dict:
+def map_to_annotation_event(specimen_data: Dict, results: List[Dict[str, str]],
+        job_id: str) -> dict:
     """
     Map the result of the API call to an annotation
     :param specimen_data: The JSON value of the Digital Specimen
     :param results: A list of results that contain the queryString and the geoCASe identifier
     :param job_id: The job ID of the MAS
     :return: Returns a formatted annotation Record which includes the Job ID
     """
-    timestamp = timestamp_now()
+    timestamp = shared.timestamp_now()
     if results is None:
         annotations = list()
     else:
-        annotations = list(map(lambda result: map_to_annotation(specimen_data, result, timestamp), results))
+        ods_agent = shared.get_agent()
+        annotations = list(
+            map(lambda result: map_result_to_annotation(specimen_data, result,
+                                                        timestamp, ods_agent),
+                results))
     mas_job_record = {
         "jobId": job_id,
         "annotations": annotations
     }
     return mas_job_record
 
 
-def map_to_annotation(specimen_data, result, timestamp):
-    oa_value = {
-        'entityRelationships': {
-            'entityRelationshipType': 'hasGeoCASeID',
-            'objectEntityIri': f'https://geocase.eu/specimen/{result["geocaseId"]}',
-            'entityRelationshipDate': timestamp,
-            'entityRelationshipCreatorName': os.environ.get('MAS_NAME'),
-            'entityRelationshipCreatorId': f"https://hdl.handle.net/{os.environ.get('MAS_ID')}"
-        }
-    }
-    annotation = {
-        'rdf:type': 'Annotation',
-        'oa:motivation': 'ods:adding',
-        'oa:creator': {
-            ODS_TYPE: 'oa:SoftwareAgent',
-            'foaf:name': os.environ.get('MAS_NAME'),
-            ODS_ID: f"https://hdl.handle.net/{os.environ.get('MAS_ID')}"
-        },
-        'dcterms:created': timestamp,
-        'oa:target': {
-            ODS_ID: specimen_data[ODS_ID],
-            ODS_TYPE: specimen_data[ODS_TYPE],
-            'oa:selector': {
-                ODS_TYPE: 'ClassSelector',
-                'oa:class': '$.entityRelationships'
-            },
-        },
-        'oa:body': {
-            ODS_TYPE: 'TextualBody',
-            'oa:value': [json.dumps(oa_value)],
-            'dcterms:reference': result['queryString']
-        }
-    }
-    return annotation
-
-
-def timestamp_now() -> str:
+def map_result_to_annotation(specimen_data: Dict, result: Dict, timestamp: str,
+        ods_agent: Dict) -> Dict:
```

Comment:
Maybe just call it agent?

Reply:
this indicates that it is a mapped annotation object, will keep it as is